scripts/managedb/attrsvector2attrsobject.py

```diff
@@ -0,0 +1,172 @@
+# iot_support at tid dot es
+
+__author__ = 'fermin'
+
+from pymongo import MongoClient
+import json
+import sys
+from time import sleep
+
+ATTRNAMES = 'attrNames'
+ATTRS     = 'attrs'
+
+# The way in which Python manage dictionaries doesn't make easy to be sure
+# of field ordering, which is important for MongoDB in the case of using an
+# embedded document for _id. The flatten() function helps
+def flatten(_id):
+    r = {'_id.id': _id['id']}
+
+    if 'type' in _id:
+       r['_id.type'] = _id['type']
+    else:
+       r['_id.type'] = {'$exists': False}
+
+    if 'servicePath' in _id:
+       r['_id.servicePath'] = _id['servicePath']
+    else:
+       r['_id.servicePath'] = {'$exists': False}
+
+    return r
+
+def update_ok(doc, n_attr_names, n_attrs):
+
+    if (not ATTRNAMES in doc or not ATTRS in doc):
+        print "debug1"
+        return False
+
+    ck_attr_names = len(doc[ATTRNAMES])
+    ck_attrs = len(doc[ATTRS].keys())
+
+    if (n_attr_names != ck_attr_names or n_attrs != ck_attrs):
+        print "debug 2: %d %d %d %d" % (n_attr_names, ck_attr_names, ck_attrs, n_attrs)
+        return False
+
+    return True
+
+if len(sys.argv) != 2:
+    print "missing db name"
+    sys.exit()
+
+DB = sys.argv[1]
+COL = 'entities'
+
+# Warn user
+print "WARNING!!!! This script modifies your '%s' database. It is STRONGLY RECOMMENDED that you" % DB
+print "do a backup of your database before using it as described in https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/Publish/Subscribe_Broker_-_Orion_Context_Broker_-_Installation_and_Administration_Guide#Backup. Use this script at your own risk."
+print "If you are sure you want to continue type 'yes' and press Enter"
+
+confirm = raw_input()
+
+if (confirm != 'yes'):
+    sys.exit()
+
+client = MongoClient('localhost', 27017)
+db = client[DB]
+
+need_fix = False
+
+skipped     = 0
+changed     = 0
+error       = 0
+processed   = 0
+
+total = db[COL].count()
+
+print "- processing entities collection (%d entities) changing attrs vectors to objects, this may take a while... " % total
+
+# The sort() is a way of ensuring that a modified document doesn't enters again at the end of the cursor (we have
+# observed that this may happen with large collections, e.g ~50,000 entities). In addition, we have to use
+# batch_size so the cursor doesn't expires at server (see http://stackoverflow.com/questions/10298354/mongodb-cursor-id-not-valid-error).
+# The used batch_size value is an heuristic
+for doc in db[COL].find().sort([('_id.id', 1), ('_id.type', -1), ('_id.servicePath', 1)]).batch_size(100):
+
+    processed += 1
+
+    sys.stdout.write('- processing entity: %d/%d   \r' % (processed, total) )
+    sys.stdout.flush()
+
+    # Check that attribute field exists and is a vector (otherwise this entity is skipped)
+    old_attrs = doc[ATTRS]
+    if not isinstance(old_attrs, list):
+        #print '- %d: not vector attribute for entity %s. Skipping' % (processed, json.dumps(doc['_id']))
```

Comment:
I think we have an issue to do an overall fix of than in all the scripts (I don't remember the number...). However, this script is so simple and for using just one time (when going from 0.20.0 to 0.21.0) that probably doesn't worth the effort.